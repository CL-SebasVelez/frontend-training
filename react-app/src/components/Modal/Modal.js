import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'emerald-ui/lib/Modal';
import Button from 'emerald-ui/lib/Button';

function DefaultModal(props) {
  return (
    <div>
      <Modal
        onHide={() => {
          props.close();
        }}
        show={props.show}
      >
        <Modal.Header closeButton={true}>
          <Modal.Title className="Hi">{props.title}</Modal.Title>
        </Modal.Header>
        <Modal.Body className="defaultRender">{props.render()}</Modal.Body>
        <Modal.Footer>
          <Modal.Footer>
            <Button onClick={props.close} shape="flat" color="info">
              Cancel
            </Button>
            <Button color="info" onClick={props.accept}>
              {props.acceptButton}
            </Button>
          </Modal.Footer>
        </Modal.Footer>
      </Modal>
    </div>
  );
}

DefaultModal.propTypes = {
  show: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  accept: PropTypes.func.isRequired,
  acceptButton: PropTypes.string.isRequired,
};

DefaultModal.defaultProps = {
  show: false,
  close: function () {
    this.show = false;
  },
  title: 'Information',
  render: () => <div className="defaultRender">There's not Information</div>,
  accept: function () {
    this.show = false;
  },
  acceptButton: 'Accept',
};

export default DefaultModal;
