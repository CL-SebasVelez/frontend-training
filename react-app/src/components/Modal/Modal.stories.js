import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Modal from './Modal';

storiesOf('Modal', module)
    .add('with title', withInfo()(() => (
        <Modal title="Modal title" />
    )));
