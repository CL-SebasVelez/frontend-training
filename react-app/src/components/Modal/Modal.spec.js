import React from 'react';
import renderer from 'react-test-renderer';

import Modal from './Modal';

describe('Modal component', () => {
  it('should render with default props', () => {
    const tree = renderer.create(<Modal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should launch error when pass wrong prop types', () => {
    const { show, close, title, render, accept, acceptButton } = 10;
    expect(() =>
      shallow(
        <Modal
          title={title}
          show={show}
          close={close}
          render={render}
          accept={accept}
          acceptButton={acceptButton}
        />
      )
    ).toThrowError();
  });

  it('should render all props as passed it', () => {
    const title = 'New Modal';
    const show = false;
    const close = () => {
      return 'closing Modal';
    };
    const render = () => <div>Render</div>;
    const accept = () => {
      return 'accepting Modal';
    };
    const acceptButton = 'Accept';
    const testRenderer = renderer.create(
      <Modal
        title={title}
        show={show}
        close={close}
        render={render}
        accept={accept}
        acceptButton={acceptButton}
      />
    );
    const testInstance = testRenderer.root;
    expect(testInstance.findByType(Modal).props.title).toBe(title);
    expect(testInstance.findByType(Modal).props.show).toBe(show);
    expect(testInstance.findByType(Modal).props.close).toBe(close);
    expect(testInstance.findByType(Modal).props.render).toBe(render);
    expect(testInstance.findByType(Modal).props.accept).toBe(accept);
    expect(testInstance.findByType(Modal).props.acceptButton).toBe(
      acceptButton
    );
  });

  it('should not show the modal when pass `show` prop as false', () => {
    const testRenderer = renderer.create(<Modal show={false} />);
    const testInstance = testRenderer.root;
    expect(testInstance.findByType(Modal).props.show).toBe(false);
  });
});
