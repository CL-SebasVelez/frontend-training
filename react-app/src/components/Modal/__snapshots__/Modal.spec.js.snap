// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render with default props 1`] = `<div />`;

exports[`Modal component should render without default props 1`] = `<div />`;
